docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -268,11 +271,19 @@ kube-dns-{network} shall be managed by Endpoints controller just like a regular
 
 __EndpointSlices__, the new type introduced in k8s v1.17, is out of current scope.
 
-
-
 ### Ingress/egress/network policies
 (**TBD: more details required**)
 
+## CNI spec
+CNI_ARGS is leveraged for Arktos workload runtime to pass information on to the cni plugin of the network type.
+
+By spec the value of CNI_ARGS env var is alphanumeric key-value pairs separated by semicolons, e.g. "IPV4=12.34.56.78;VPCID=mizar-12345".
```

Comment:
Just curious; Is there a mechanism to update CNI_ARGS (after provisioning the PODs)? If not that's fine, I just need to know.

Reply:
cni invocation a one-time call and cni process runs and exit. 
k8s does not handle annotation changes in our case and no CNI_ARGS  updates would be detected, hence no mechanism to automate the pod re-creation. If we do think such a mechanism is critical and we should support,  we need to put such info in spec fields instead of annotations.
manual stopping pod and starting pod will bring pod with the new cni-args; if such mechanism exist, it is essentially a similar restarting.

Reply:
I think it is necessary. Here are two scenarios in which we need to update (re)-invoke the CNI some how:
1. In Mizar we update the list of bouncers (transit switch) to the host.
1. The user add/remove NICs from a POD.

In Mizar, we can do these updates with RPC calls to the transit daemon. But that's something Mizar specific. If CNI and runtime workflow is not capable of doing these updates, then how this solution supports different network provider?

Reply:
@zasherif  I've change the design to allow only one nic each pod. Let's focus on single nic pod right now, defer the consideration of multiple nics support to later.

For reason 1, Mizar updates the list of bouncer internally, will the nic already allocated be affected? will its ip address change?  Can we make _encapsulate_ Mizar change internal to Mizar and not affect the nic that has already been allocated to pod?

Reply:
exchanged emails offline. It turns out a Mizar operator will do (most of the) underlying network wirings based on  network_user_input annotation of pod; its cni is mere a synchronization mechanism to wait for the network to be provisioned. Arktos won't prevent such interaction sequence, with current design. Such interactions would be primarily driven by Mizar network controller itself.